pytealutils/iter/iter_test.py

```diff
@@ -0,0 +1,44 @@
+from iter import iterate
+from pyteal import (
+    Bytes,
+    Expr,
+    Int,
+    Itob,
+    Log,
+    Mode,
+    ScratchVar,
+    Seq,
+    Subroutine,
+    SubroutineCall,
+    TealType,
+    compileTeal,
+)
+
+
+def test_iterate():
+    res = iterate(Log(Bytes("a")), Int(10))
+    assert type(res) is SubroutineCall
+    assert res is not None
+
+    src = compile_app(res)
+    assert len(src) > 0
+
```

Comment:
So here's a stab at making this into a unit tests that builds up PyTEAL in two different ways and then asserts that the generated TEAL is the same. For this to work you need something like the `TealDiff` class below, which I have defined in [this gist](https://gist.github.com/tzaffi/634d98dd92547566417137b509d92052).

Not sure how robust this approach is, but at least its something.

```suggestion

    # build expected_src using pure PyTEAL:
    i = ScratchVar()
    expected_src = compile_app(
        SubroutineDefinition(
            lambda: For(
                i.store(Int(0)),
                i.load() < Int(10),
                i.store(i.load() + Int(1)),
            ).Do(Log(Bytes("a"))),
            TealType.none,
        ).invoke([])
    )

    diff = TEALDiff(src, expected_src)
    diff.assert_equals(msg=diff.first_diff())
```